src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.VsCodeWindowManager.cs

```diff
@@ -224,6 +231,70 @@ public int RemoveAdornments()
 
                 return VSConstants.S_OK;
             }
+
+            // GetOutline is called every time a new code window is created. Whenever we switch to a different window, it is guaranteed
+            // that ReleaseOutline will be called on the old window before GetOutline is called for the new window. 
```

Comment:
Was that guarantee coming from documentation somewhere, or did we confirm it internally with somebody?

Reply:
Apparently there is existing functionality in the codebase that relies on this assumption holding true. I believe @jmarolf can elaborate further :)